src/core/node_descriptors.rs

```diff
@@ -55,6 +55,13 @@ impl<'repo> NodeObject<'repo> {
             NodeObject::GarbageCollected { oid } => *oid,
         }
     }
+
+    fn get_short_oid(&self) -> eyre::Result<String> {
+        match self {
+            NodeObject::Commit { commit } => Ok(commit.get_short_oid()?),
+            NodeObject::GarbageCollected { oid } => Ok(oid.to_string()[..8].to_string()),
```

Comment:
After this, every commit id in snapshots is shorted. (length `8 -> 7`). And I think it is strange

Reply:
I believe that's the point: it means that it's working correctly! 

The docs for [`short_id`](https://docs.rs/git2/0.8.0/git2/struct.Object.html#method.short_id) say:
> Get a short abbreviated OID string for the object
> 
> This starts at the "core.abbrev" length (default 7 characters) and iteratively extends to a longer string if that length is > ambiguous. The result will be unambiguous (at least until new objects are added to the repository).

Reply:
Yes you are totally right. I thought it will generate something short but still 8+ length. I should have check docs. Thanks for sharing.